### PR TITLE
[bug] Canary-check: Pod: Issue with scheduling pods in case taints and tolerations using #1232

### PR DESCRIPTION
**Pull request by msergg**
https://github.com/flanksource/canary-checker/issues/1232

### CODE REVIEW

**Comment by moshloop** [comment]
Hi @msergg and thank you for the contribution 

The reasoning behind this behaviour is to test the schedulability of every node in a cluster - mostly relevant for on-premise / bare-metal clusters - however I do agree that it is perhaps not the best default behaviour, could you gate the old behaviour in a `roundRobinNodes` option?

**Comment by msergg** [comment]
> roundRobinNodes

Do you mean that there is an option in documentation for roundRobinNodes? Or need to develop this option?


About default behaviour, a also got errors during scheduling on both premise and cloud clusters,
in case we have some topology constraints or worse - some selected nodes could be already full and scheduler could not deploy test pod to this insance - so check will fail and it not show correct cluster scheduling state


I thing better to remove such behaviour or configure it to seperate option if some one need it

**Comment by moshloop** [comment]
@msergg - The `roundRobinNodes` option doesn't exist, I am suggesting that you add it as part of your PR

**Comment by msergg** [comment]
@moshloop 
I have added an option usePreparedNodeSheduling
how do you test  it locally if shema changes? do you have any procedure?

**Comment by moshloop** [comment]
@msergg If you run `make resources` it will regenerate all the CRD's and manifest files

**Comment by msergg** [comment]
@moshloop pls check changes , seem this logic should be enough

File: checks/pod.go [hunk 1]
```diff
@@ -89,9 +91,13 @@ func (c *PodChecker) newPod(podCheck canaryv1.PodCheck, nodeName string) (*v1.Po
 	pod.Labels[nameLabel] = pod.Name
 	pod.Labels[podCheckSelector] = c.podCheckSelectorValue(podCheck)
 	pod.Labels[podGeneralSelector] = "true"
-	pod.Spec.NodeSelector = map[string]string{
-		"kubernetes.io/hostname": nodeName,
+
+	if podCheck.RoundRobinNodes == false {
```
Comment (moshloop) [inline checks/pod.go:95]:
```suggestion
	if podCheck.RoundRobinNodes  {
```

**Comment by msergg** [comment]
cannot pass all license/cla , some commits were pushed from another unexisting users

**Comment by msergg** [comment]
@moshloop could you pls approve next steps or we need to fix smths?

**Comment by moshloop** [comment]
@msergg If you squash all your changes under `msergg` and not `serhii.martyniuk` the CLA check will pass

**Comment by msergg** [comment]
@moshloop could you pls review?

**Pull request merged**